tests/transforms/test_classification.py

```diff
@@ -0,0 +1,417 @@
+        with pytest.raises(ValueError) as e:
+            main(InferenceParser(input_data=input_data, args=[]))
+        assert "Expected the number of predictions" in str(e.value)
+
+    def test_munge_data(self, input_data, traces, rois, metadata):
+        parser = InferenceParser(input_data=input_data, args=[])
+        roi_data = RoiJsonSchema(many=True).load(rois)
+        expected_rois = [coo_matrix(np.array(r["mask_matrix"])) for r in rois]
+        actual_rois, actual_metadata, actual_traces, actual_np_traces = (
+            _munge_data(parser, roi_data))
+        for ix, e_roi in enumerate(expected_rois):
+            np.testing.assert_array_equal(
+                e_roi.toarray(), actual_rois[ix].toarray(),
+                err_msg=("Expected ROIs did not equal actual ROIs from "
+                         f" _munge_data for index={ix}."))
+        np.testing.assert_array_equal(
+            traces, actual_traces,
+            err_msg=("Expected traces did not equal actual traces "
+                     "from _munge_data"))
+        np.testing.assert_array_equal(
+            traces, actual_np_traces,
+            err_msg=("Expected neuropil traces did not equal actual neuropil "
+                     "traces from _munge_data"))
+        assert [metadata]*len(rois) == actual_metadata
+
+
+def test_InferenceInputSchema_classifier_model_path(
+        classifier_model, s3_classifier, input_data, tmp_path):
+    ArgSchemaParser(
+        input_data=input_data,      # input_data has local classifier default
+        schema_type=InferenceInputSchema,
+        args=[])
+    s3_input = input_data.copy()
+    s3_input["classifier_model_path"] = s3_classifier
+    ArgSchemaParser(
+        input_data=s3_input,
+        schema_type=InferenceInputSchema,
+        args=[])
+    fake_s3 = input_data.copy()
+    fake_s3["classifier_model_path"] = "s3://my-bucket/fake-hello.txt"
+    with pytest.raises(ValidationError) as e:
+        ArgSchemaParser(
+            input_data=fake_s3,
+            schema_type=InferenceInputSchema,
+            args=[])
+    assert "does not exist" in str(e.value)
+    fake_local = input_data.copy()
+    fake_local["classifier_model_path"] = str(tmp_path / "hello-again.txt")
+    with pytest.raises(ValidationError) as e:
+        ArgSchemaParser(
+            input_data=fake_local,
+            schema_type=InferenceInputSchema,
+            args=[])
+    assert "does not exist" in str(e.value)
+
+
+def test_InferenceInputSchema_fails_invalid_roi(
+        input_data, bad_roi_json):
+    input_data.update({"roi_masks_path": bad_roi_json})
+    with pytest.raises(ValidationError) as e:
+        parser = ArgSchemaParser(
+            input_data=input_data,
+            schema_type=InferenceInputSchema,
+            args=[])
+        main(parser)
+    assert "Missing data for required field" in str(e.value)
+    assert "Not a valid integer" in str(e.value)
+
+
+class TestRoiJsonSchema:
```

Comment:
why are these inside the class? I don't see any reason they can't just be pulled out and stand alone.

Reply:
For clarity/organization purposes. Usually I wouldn't have multiple classes per file, so the tests are more obvious.

Really I should have organized the InferenceInputSchema tests into a test case, but that was the first thing I wrote so I was running all the tests in the file with every save anyway. I'd actually like to go ahead and do that. Are you ok with that?

Reply:
i'm fine with that. I figured it was for organization, but I wanted to check if there was some intended functional reason I wasn't catching.